core-blocks/video/edit.js

```diff
@@ -123,6 +124,17 @@ class VideoEdit extends Component {
 							onChange={ this.toggleAttribute( 'controls' ) }
 							checked={ controls }
 						/>
+						<SelectControl
+							label={ __( 'Preload' ) }
+							value={ undefined !== preload ? preload : 'metadata' }
```

Comment:
Would setting a default value for the attribute definition solve the need for this undefined checks?
On the save we would only set the attribute if it was different from the default.

Reply:
Thanks for your feedback, thats a better solution. I committed the change.